crates/sui-json-rpc/src/quorum_driver_api.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::api::QuorumDriverApiServer;
+use crate::SuiRpcModule;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult;
+use jsonrpsee_core::server::rpc_module::RpcModule;
+use move_bytecode_utils::module_cache::SyncModuleCache;
+use signature::Signature;
+use std::sync::Arc;
+use sui_core::authority::{AuthorityStore, ResolverWrapper};
+use sui_core::authority_client::NetworkAuthorityClient;
+use sui_json_rpc_types::SuiExecuteTransactionResponse;
+use sui_open_rpc::Module;
+use sui_quorum_driver::QuorumDriver;
+use sui_types::crypto::SignatureScheme;
+use sui_types::messages::{ExecuteTransactionRequest, ExecuteTransactionRequestType};
+use sui_types::sui_serde::Base64;
+use sui_types::{
+    crypto,
+    crypto::SignableBytes,
+    messages::{Transaction, TransactionData},
+};
+
+pub struct FullNodeQuorumDriverApi {
+    pub quorum_driver: Arc<QuorumDriver<NetworkAuthorityClient>>,
+    pub module_cache: Arc<SyncModuleCache<ResolverWrapper<AuthorityStore>>>,
```

Comment:
This one makes me a bit nervous, as it makes quorum driver dependent on the store.
 Do we have a concrete use case where tx sender wants to see events in json format?

Reply:
good point. I don't really know but has increasingly started to think we just need bcs in the responses. Let's make an issue for this and move the discussion there - if we want to remove the parsed structs, we should phase them out entirely. cc @patrickkuo